JsonSchema.Generation/SchemaGeneratorConfiguration.cs

```diff
@@ -34,11 +35,42 @@ public class SchemaGeneratorConfiguration
 	/// <remarks>
 	/// This can be replaced with any `Func&lt;string, string&gt;`.
 	/// </remarks>
-	public PropertyNamingMethod PropertyNamingMethod
+	[Obsolete($"Use {nameof(PropertyNameResolver)} instead.")]
+	public PropertyNamingMethod? PropertyNamingMethod
 	{
-		get => _propertyNamingMethod ??= PropertyNamingMethods.AsDeclared;
+		get => _propertyNamingMethod;
 		set => _propertyNamingMethod = value;
 	}
```

Comment:
Instead of all of the conditional logic in `PropertyNameResolver`, this should actually be implemented _using_ `PropertyNameResolver`.  Something like this:

```c#
[Obsolete($"Use {nameof(PropertyNameResolver)} instead.")]
public PropertyNamingMethod? PropertyNamingMethod
{
	get => x => PropertyNameResolver(new DummyInfo(x));
	set => PropertyNameResolver = x => x.Name;
}

private class DummyInfo : MemberInfo
{
	public override object[] GetCustomAttributes(bool inherit) => Array.Empty<object>();

	public override object[] GetCustomAttributes(Type attributeType, bool inherit) => Array.Empty<object>();

	public override bool IsDefined(Type attributeType, bool inherit) => false;

	public override Type DeclaringType { get; } = typeof(DummyInfo);
	public override MemberTypes MemberType { get; } = MemberTypes.Property;
	public override string Name { get; }
	public override Type? ReflectedType { get; } = null;

	public DummyInfo(string name)
	{
		Name = name;
	}
}
```

Then `PropertyNameResolver` can be just an autoproperty.

Since this library isn't calling `PropertyNamingMethod` anymore, its getter _should_ never be called, so we don't need to worry about it instantiating a bunch of objects.

Reply:
done that, it's now a simple autoproperty